internal/pkg/agent/application/upgrade/step_download.go

```diff
@@ -33,7 +35,7 @@ const (
 	defaultUpgradeFallbackPGP = "https://artifacts.elastic.co/GPG-KEY-elastic-agent"
 )
 
-func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string, skipVerifyOverride bool, skipDefaultPgp bool, pgpBytes ...string) (_ string, err error) {
+func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string, details *details.Details, skipVerifyOverride bool, skipDefaultPgp bool, pgpBytes ...string) (_ string, err error) {
```

Comment:
Why are we passing the details here if they are not used by this function ?

Reply:
Good catch, thanks! We do actually need the details here so the download progress percentage can be set in the details metadata field.  Looks like I just forgot to implement that bit so let me work on that.

Reply:
I'm going to need to refactor [`downloadProgressReporter`](https://github.com/elastic/elastic-agent/blob/7f842a32410fb0f9f4fa1c3cdecb7b6717d0b17d/internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go#L223) a little bit to work with the `details` object here in addition to the logging it's already doing. Are you okay with me adding all those changes to this PR or would you prefer them in a follow up PR?

Reply:
This refactor is turning out to be a rather sizeable change so I'll do it in a separate PR.  Once that PR is reviewed and merged, I'll update this PR here to use it.

Reply:
Refactoring PR is up: https://github.com/elastic/elastic-agent/pull/3542.